Ductus.FluentDocker/Builders/ContainerBuilder.cs

```diff
@@ -425,14 +436,33 @@ public ContainerBuilder ReuseIfExists()
     /// assignment. The rest of the networks will do attach via docker network.
     /// </remarks>
     public ContainerBuilder UseNetwork(params INetworkService[] network)
+    {
+      return UseNetwork(null, network);
+    }
+
+    /// <summary>
+    ///   Uses a already pre-existing network service. It will automatically
+    ///   detach this container from the network when the network is disposed.
+    /// </summary>
+    /// <param name="network">The networks to attach this container to.</param>
+    /// <param name="alias">The alias to use when attaching this container to the given networks.</param>
+    /// <returns>Itself for fluent access.</returns>
+    /// <remarks>
+    /// The firstg parameter will not be used to do a docker network attach. Instead
+    /// it is used as a creation parameter --network. This is to support static ip
+    /// assignment. The rest of the networks will do attach via docker network.
+    /// </remarks>
+    public ContainerBuilder UseNetwork(string alias, params INetworkService[] network)
     {
       if (null == network || 0 == network.Length)
         return this;
 
       if (null == _config.Networks)
-        _config.Networks = new List<INetworkService>();
+        _config.Networks = new List<NetworkAttachConfiguration>();
 
-      _config.Networks.AddRange(network);
+      
+      _config.Networks.AddRange(
+        network.Select(n => new NetworkAttachConfiguration { Alias = alias, Network = n}));
```

Comment:
I'm not familiar with alias, how does it work? Isn't alias for one single network and not for all networks?